src/lava_callback.py

```diff
@@ -56,14 +59,34 @@ def hello():
 
 @app.post('/node/<node_id>')
 def callback(node_id):
+    settings = kernelci.settings.Settings(SETTINGS_FILE)
+    tokens = settings.get(SETTINGS_PREFIX, 'secrets', 'token')
+    if not tokens:
+        return 'Unauthorized', 401
     data = request.get_json()
     job_callback = kernelci.runtime.lava.Callback(data)
 
     api_config_name = job_callback.get_meta('api_config_name')
-    api_token = request.headers.get('Authorization')
+    lab_token = request.headers.get('Authorization')
+    # return 401 if no token
+    if not lab_token:
+        return 'Unauthorized', 401
+    # iterate over tokens and check if value of one matches
+    # retrieve lab name from key
+    lab_name = None
+    for token in tokens:
+        if tokens[token] == lab_token:
+            lab_name = tokens[token]
```

Comment:
Naming here is a bit confusing: `tokens` keeps `{'lab-name': 'lab-token'}` dict. How about:

```suggestion
    for lab, token in tokens.items():
        if token == lab_token:
            lab_name = lab
```

(or at least change assignment before lookup break to `token` - otherwise token value gets assigned instead of lab name)

Reply:
I updated code in this part with new approach (as we might have 2 tokens)